### PR TITLE
CDL Datatype definition resource and data sources added

### PR DESCRIPTION
**Pull request by roopesh83**

### CODE REVIEW

File: hsdp/provider.go [hunk 1]
```diff
@@ -193,6 +193,7 @@ func Provider(build string) *schema.Provider {
 			"hsdp_dicom_gateway_config":      resourceDICOMGatewayConfig(),
 			"hsdp_cdl_research_study":        resourceCDLResearchStudy(),
 			"hsdp_dicom_remote_node":         resourceDICOMRemoteNode(),
+			"hsdp_cdl_datatypedefinition":	  resourceCDLDataTypeDefinition(),
```
Comment (loafoe) [inline hsdp/provider.go:196]: Please use `hsdp_cdl_data_type_definition`

File: hsdp/provider.go [hunk 2]
```diff
@@ -220,6 +221,8 @@ func Provider(build string) *schema.Provider {
 			"hsdp_cdl_research_study":          dataSourceCDLResearchStudy(),
 			"hsdp_cdl_research_studies":        dataSourceCDLResearchStudies(),
 			"hsdp_container_host_instances":    dataSourceContainerHostInstances(),
+			"hsdp_cdl_datatype_definitions":	dataSourceCDLDataTypeDefinitions(),
```
Comment (loafoe) [inline hsdp/provider.go:224]: same here for consistency `hsdp_cdl_data_type_definitions`

File: hsdp/resource_cdl_datatype_definition.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"jsonschema": {
```
Comment (loafoe) [inline hsdp/resource_cdl_datatype_definition.go:38]: `json_schema`

File: hsdp/resource_cdl_datatype_definition.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	}
+	_ = d.Set("name", dataTypeDefinition.Name)
+	_ = d.Set("description", dataTypeDefinition.Description)
+
+	b, err := json.Marshal(dataTypeDefinition.JsonSchema)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_ = d.Set("jsonschema", string(b))
```
Comment (loafoe) [inline hsdp/resource_cdl_datatype_definition.go:156]: Use underscore naming here

**Review by loafoe — CHANGES_REQUESTED** [review]

**Review by loafoe — APPROVED** [review]

File: hsdp/data_source_cdl_datatype_definition.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+		Schema: map[string]*schema.Schema{
+			"cdl_endpoint": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"dtd_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"jsonschema": {
```
[the remark this answers is not in the document]
Comment (roopesh83) [reply hsdp/data_source_cdl_datatype_definition.go]: Renamed to json_schema

File: hsdp/data_source_cdl_datatype_definition.go [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"createdby": {
```
[the remark this answers is not in the document]
Comment (roopesh83) [reply hsdp/data_source_cdl_datatype_definition.go]: renamed to created_by

File: hsdp/data_source_cdl_datatype_definition.go [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"createdon": {
```
[the remark this answers is not in the document]
Comment (roopesh83) [reply hsdp/data_source_cdl_datatype_definition.go]: createdon -> created_on (renamed)

File: hsdp/data_source_cdl_datatype_definition.go [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updatedby": {
```
[the remark this answers is not in the document]
Comment (roopesh83) [reply hsdp/data_source_cdl_datatype_definition.go]: updatedby -> updated_by (renamed)

File: hsdp/data_source_cdl_datatype_definition.go [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updatedon": {
```
[the remark this answers is not in the document]
Comment (roopesh83) [reply hsdp/data_source_cdl_datatype_definition.go]: updatedon -> updated_on (renamed)

File: hsdp/data_source_cdl_datatype_definition.go [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+		return diag.FromErr(err)
+	}
+
+	b, err := json.Marshal((*dataTypeDefinition).JsonSchema)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId((*dataTypeDefinition).ID)
+	_ = d.Set("jsonschema", string(b))
```
[the remark this answers is not in the document]
Comment (roopesh83) [reply hsdp/data_source_cdl_datatype_definition.go]: using underscore names like json_schema, created_on ...

File: hsdp/provider.go (hunk 1 above)
Comment (roopesh83) [reply hsdp/provider.go]: hsdp_cdl_datatypedefinition -> hsdp_cdl_data_type_definition (renamed)

Comment (roopesh83) [reply hsdp/provider.go]: Looks like I missed "go fmt" for this script, will format it.

File: hsdp/provider.go (hunk 2 above)
Comment (roopesh83) [reply hsdp/provider.go]: hsdp_cdl_datatype_definitions -> hsdp_cdl_data_type_definitions
hsdp_cdl_datatype_definition -> hsdp_cdl_data_type_definition

File: hsdp/resource_cdl_datatype_definition.go (hunk 3 above)
Comment (roopesh83) [reply hsdp/resource_cdl_datatype_definition.go]: renamed

File: hsdp/resource_cdl_datatype_definition.go (hunk 4 above)
Comment (roopesh83) [reply hsdp/resource_cdl_datatype_definition.go]: renamed to json_schema

**Pull request merged**